src/tracer/MOM_neutral_diffusion.F90

```diff
@@ -877,16 +946,292 @@ subroutine find_neutral_surface_positions(nk, Pl, Tl, Sl, dRdTl, dRdSl, Pr, Tr,
       hL = absolute_position(nk,Pl,KoL,PoL,k_surface) - absolute_position(nk,Pl,KoL,PoL,k_surface-1)
       hR = absolute_position(nk,Pr,KoR,PoR,k_surface) - absolute_position(nk,Pr,KoR,PoR,k_surface-1)
       if ( hL + hR > 0.) then
-        hEff(k_surface-1) = 2. * hL * hR / ( hL + hR )
+        hEff(k_surface-1) = 2. * hL * hR / ( hL + hR ) ! Harmonic mean of layer thicknesses
       else
         hEff(k_surface-1) = 0.
       endif
     endif
 
   enddo neutral_surfaces
 
-end subroutine find_neutral_surface_positions
+end subroutine find_neutral_surface_positions_continuous
+
+!> Higher order version of find_neutral_surface_positions. Returns positions within left/right columns
+!! of combined interfaces using intracell reconstructions of T/S
+subroutine find_neutral_surface_positions_discontinuous(nk, Pres_l, Tl, Sl, dRdT_l, dRdS_l, &
```

Comment:
It's a shame we can't reuse the same algorithm, or combine them. Debugging these sorts of algorithms is painful. Is this the same algorithm but with different data structures being passed in?

One test to try is to set the discontinuous edge values to the interface values of the "continuous" approach. It should give the same solution as the continuous but with 2*G%ke vanished sub-layers.